app/questionnaire/navigator.py

```diff
@@ -0,0 +1,182 @@
+
+    :param blocks: A list containing all blocks in the survey
+    :param block_id: The block id to visit
+    :param answers: The answers to use on evaluating rules
+    :param path: The known path as a list which has been visited already
+    :return: A list of block ids followed through the survey
+    """
+    if block_id in Navigator.CLOSING_INTERSTITIAL_PATH:
+        return path
+
+    path.append(block_id)
+
+    # Return the index of the block id to be visited
+    block_id_index = next(index for (index, b) in enumerate(blocks) if b["id"] == block_id)
+
+    if 'routing_rules' in blocks[block_id_index]:
+        for rule in blocks[block_id_index]['routing_rules']:
+            goto_rule = rule['goto']
+            if 'when' in goto_rule.keys():
+                answer_index = goto_rule['when']['id']
+                if answer_index in answers:
+                    answer = answers[answer_index]
+                    if evaluate_rule(goto_rule, answer):
+                        return visit_block(blocks, goto_rule['id'], answers, path)
+                else:
+                    return path
+            elif 'id' in goto_rule.keys():
+                return visit_block(blocks, goto_rule['id'], answers, path)
+    else:
+        if block_id_index != len(blocks) - 1:
+            next_block_id = blocks[block_id_index + 1]['id']
+            return visit_block(blocks, next_block_id, answers, path)
+    return path
+
+
+class Navigator:
+    PROCEEDING_INTERSTITIAL_PATH = ['introduction']
+    CLOSING_INTERSTITIAL_PATH = ['summary', 'thank-you']
+
+    def __init__(self, survey_json):
+        self.survey_json = survey_json
+        self.blocks = self.get_blocks()
+        self.first_block_id = self.blocks[0]['id']
+
+    def get_routing_path(self, answers=None):
+        """
+        Returns a list of the block ids visited based on answers provided
+        :return: List of block ids
+        """
+        answers = answers or {}
+        routing_path = []
+
+        return visit_block(self.blocks, self.first_block_id, answers, routing_path)
+
+    def get_location_path(self, answers=None):
+        """
+        Returns a list of url locations visited based on answers provided
+        :param answers:
+        :return:
+        """
+        answers = answers or {}
+        routing_path = visit_block(self.blocks, self.first_block_id, answers, [])
+        # Make sure we don't update original
+        location_path = list(Navigator.PROCEEDING_INTERSTITIAL_PATH)
+        location_path.extend(routing_path)
+        location_path.extend(Navigator.CLOSING_INTERSTITIAL_PATH)
+
+        return location_path
+
+    def get_first_block_id(self):
+        return self.survey_json['groups'][0]['blocks'][0]['id']
+
+    def get_blocks(self):
+        blocks = []
+        for group in self.survey_json['groups']:
+            blocks.extend([block for block in group['blocks']])
+        return blocks
+
+    def get_block_ids(self):
```

Comment:
There's no test coverage on this function; which is suspicious, would have thought other tests would have hit it indirectly, is this function redundant ?

https://codecov.io/gh/ONSdigital/eq-survey-runner/compare/b9b82d14406a662b6c8bede15bc3ee72e18347c4...333b0e5b685e2fab03a8eb7ad9cdaeac72c01c1a#6170702F7175657374696F6E6E616972652F6E6176696761746F722E7079-108